source/dh_overrides.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Changes to OpenSSL version 1.1.1.
+ * Copyright Amazon.com, Inc. All Rights Reserved.
+ * Copyright 1995-2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <cbmc_proof/nondet.h>
+#include <openssl/dh.h>
+#include <openssl/ossl_typ.h>
+
+int DH_size(const DH *dh) {
+    return nondet_int();
+}
+
+void DH_free(DH *dh) {
```

Comment:
nothing to check here?

Reply:
I don't think so https://www.openssl.org/docs/man1.1.0/man3/DH_free.html

Reply:
can check that `dh` is either `null` or `__CPROVER_w_ok(dh, sizeof(*dh))`

Reply:
Even better: make a `openssl_DH_is_valid()`, and use it in the precondtions. For now, it might only assert `__CPROVER_w_ok`

Reply:
Implemented as suggested.

Reply:
do it here for free as well `assert(dh == NULL || dh_is_valid())`